lib/meta.nix

```diff
@@ -132,8 +132,7 @@ rec {
         { shortName = licstr; }
       );
 
-  /* Get the path to the main program of a derivation with either
-     meta.mainProgram or pname or name
+  /* Get the path to the main program of a derivation meta.mainProgram
```

Comment:
I don't think the doc should include deprecated behaviour?

Reply:
```suggestion
  /* Get the path to the main program of a package based on meta.mainProgram
```

- added "based on"
- derivation -> package, because the builder is also a main program

Derivations are just how we build packages, and the attrset returned by `mkDerivation` is a representation of the package in context.